### ISSUE TITLE
`mergeStates` error

### ISSUE THREAD

**Issue by brodieG** [brodieG/unitizer#199]
```
> library(unitizer)
> unitize_dir()

Warning: namespace ‘ggplot2’ is not available and has been replaced             
by .GlobalEnv when processing object ‘mustWork’
| Summary of files in common directory 'unitizer':                              
| 
|                  Pass Fail
| *1.  abstract.R    20    9
| *2.     alike.R   160    1
|  3.   classes.R    23    -
| *4. datatable.R     -    7
|  5.  internal.R   134    -
|  6.  language.R    45    -
|  7.      type.R    38    -
| ..........................
|                   420   17
| Legend:
| * `unitizer` requires review

| Type number of unitizer to review, 'A' to review all that require review

unitizer> 2
+------------------------------------------------------------------------------+
| unitizer for: unitizer/alike.R                                               |
+------------------------------------------------------------------------------+

                           Pass Fail
                   Atomic    12    -
                    lists    10    -
 NULL values as wildcards     5    -
          Matrix / Arrays    10    1
              Data Frames    15    -
              Time Series     6    -
                  Factors     5    -
 Environments / Pairlists    15    -
         Calls / Formulas    37    -
                Functions     3    -
                   .alike    12    -
                 Examples    30    -
....................................
                            160    1

= Matrix / Arrays ==============================================================

- Failed -----------------------------------------------------------------------

| The following test failed because the new evaluation does not match the 
| reference value from the store. Overwrite with new result ([Y]es, [N]o, 
| [P]rev, [B]rowse, [R]erun, [Q]uit, [H]elp)?

> alike(matrix(integer(), nrow = 3, dimnames = list(c("R", "G", 
+     "B"), NULL)), matrix(1:9, nrow = 3, dimnames = list(NULL, 
+     c("R", "G", "B"))))
[1] "row.names(matrix(1:9, nrow = 3, dimnames = list(NULL, c(\"R\", \n    \"G\",
\"B\"))))\nshould be type \"character\" (is \"NULL\")"

| Value mismatch: 

< .ref                                   > .new                                 
@@ 1 @@                                  @@ 1 @@                                
< [1] "  row.names(matrix(1:9, nrow = 3  > [1] "row.names(matrix(1:9, nrow = 3, 
: , dimnames = list(NULL, c(\"R\", \n    : dimnames = list(NULL, c(\"R\", \n    
:     \"G\", \"B\"))))\nshould be type   : \"G\", \"B\"))))\nshould be type \"ch
: \"character\" (is \"NULL\")"           : aracter\" (is \"NULL\")"             

| State mismatch; see `.DIFF$state` for details.

unitizer> cat(.new, "\n")
row.names(matrix(1:9, nrow = 3, dimnames = list(NULL, c("R", 
    "G", "B"))))
should be type "character" (is "NULL") 
unitizer> Q

| No changes recorded.
Error in mergeStates(items.ref, x@state.new, x@state.ref) : 
  Logic Error: global index mapping matrix malformed; contact maintainer.
| Unexpectedly exited before storing unitizer; if you were reviewing a unitizer 
| changes to that unitizer were not saved.  Note that any unitizers you *
| completed* review of should have been saved.

> packageVersion('unitizer')
[1] ‘1.3.0.9000’
```

**Comment by brodieG** [comment]
See also the now closed issue #212

**Comment by brodieG** [comment]
Currently can't reproduce this problem.  Tried using the alike data.table tests by adding a call to `Sys.time()` to force a failure in case having a failure was a requisite condition, but no dice.  Tried both with `unitize` and `unitize_dir`.
